frontend/src/bundles/home/components/main-content/main-content.tsx

```diff
@@ -3,27 +3,65 @@ import {
     Loader,
     Overlay,
 } from '~/bundles/common/components/components.js';
-import { DataStatus } from '~/bundles/common/enums/enums.js';
+import { SocketContext } from '~/bundles/common/context/socket.js';
+import { AvatarVideoEvent, DataStatus } from '~/bundles/common/enums/enums.js';
 import {
     useAppDispatch,
     useAppSelector,
+    useContext,
     useEffect,
 } from '~/bundles/common/hooks/hooks.js';
+import { notificationService } from '~/bundles/common/services/services.js';
 import { actions as homeActions } from '~/bundles/home/store/home.js';
 
+import {
+    VIDEO_RENDER_FAILED_NOTIFICATION_ID,
+    VIDEO_RENDER_SUCCESS_NOTIFICATION_ID,
+} from '../../constants/constants.js';
+import { NotificationMessage, NotificationTitle } from '../../enums/enums.js';
 import { VideoSection } from '../components.js';
 import styles from './styles.module.css';
 
 const MainContent: React.FC = () => {
     const dispatch = useAppDispatch();
+    const socket = useContext(SocketContext);
 
     const { videos, dataStatus } = useAppSelector(({ home }) => home);
 
     // TODO: filter videos to get recent videos
-
     useEffect(() => {
-        void dispatch(homeActions.loadUserVideos());
-    }, [dispatch]);
+        const loadUserVideos = (): void => {
```

Comment:
functions in component should start with handle, please change in the whole file